src/java.base/share/classes/java/net/InetAddress.java

```diff
@@ -704,6 +706,8 @@ private static String getHostFromNameService(InetAddress addr, boolean check) {
      *
      * @return  the raw IP address of this object.
      */
+    // NOTE: This should *not* be annotated @Nullable:
+    // Both concrete implementations override it to return a non-null value.
```

Comment:
`InetAddress` isn't abstract, so somebody could instantiate and call this method and NPE. Ah, but the constructor is package-private and there are no factories.
Still, couldn't somebody have their own subclass that doesn't override this method?
Is your argument that this class should be abstract or these two methods should throw an exception instead of returning null?
The javadoc could be clearer about how this class is supposed to be used...